src/tools/synth_node.rs

```diff
@@ -0,0 +1,85 @@
+use std::{
+    io,
+    net::{IpAddr, SocketAddr},
+};
+
+use pea2pea::{protocols::Writing, Pea2Pea};
+use tokio::{
+    sync::{mpsc, mpsc::Receiver, oneshot},
+    time::timeout,
+};
+
+use crate::{
+    protocol::codecs::binary::{BinaryMessage, Payload},
+    tools::{constants::EXPECTED_MESSAGE_TIMEOUT, inner_node::InnerNode},
+};
+
+pub struct SyntheticNode {
+    inner: InnerNode,
+    receiver: Receiver<(SocketAddr, BinaryMessage)>,
+}
+
+impl SyntheticNode {
+    pub async fn new(config: pea2pea::Config) -> Self {
+        let (sender, receiver) = mpsc::channel(100);
```

Comment:
this one should probably be a `const` too

Reply:
Done